workers/loc.api/sync/get-positions-snapshot/index.js

```diff
@@ -0,0 +1,296 @@
+      }
+
+      return accum
+    }, [])
+}
+
+const _getPositionsHistoryIds = (positionsHistory) => {
+  return positionsHistory.reduce(
+    (accum, { id } = {}) => {
+      if (Number.isInteger(id)) {
+        accum.push(id)
+      }
+
+      return accum
+    }, [])
+}
+
+const _getPositionsWithActualPrice = async (
+  dao,
+  auth,
+  positions
+) => {
+  const res = []
+
+  for (const position of positions) {
+    const { mtsUpdate, symbol } = { ...position }
+
+    if (
+      !Number.isInteger(mtsUpdate) ||
+      typeof symbol !== 'string'
+    ) {
+      res.push({ ...position, actualPrice: null })
+
+      continue
+    }
+
+    const trades = await dao.findInCollBy(
+      '_getTrades',
+      {
+        auth,
+        params: {
+          symbol,
+          end: mtsUpdate,
+          limit: 1
+        }
+      }
+    )
+
+    if (
+      !Array.isArray(trades) ||
+      trades.length === 0 ||
+      !trades[0] ||
+      typeof trades[0] !== 'object' ||
+      !Number.isFinite(trades[0].execPrice)
+    ) {
+      res.push({ ...position, actualPrice: null })
+
+      continue
+    }
+
+    res.push({
+      ...position,
+      actualPrice: trades[0].execPrice
+    })
+  }
+
+  return res
+}
+
+const _filterDuplicate = (accum = [], curr = []) => {
+  if (
+    !Array.isArray(accum) ||
+    accum.length === 0
+  ) {
+    return [...curr]
+  }
+
+  const keys = Object.keys(accum[0]).filter(key => !/^_/.test(key))
+
+  return curr.filter(currItem => {
+    return accum.every(accumItem => {
+      return keys.some(key => {
+        return accumItem[key] !== currItem[key]
+      })
+    })
+  })
+}
+
+const _getPositionsAudit = async (
+  rService,
+  {
+    auth = {},
+    params: { ids } = {}
+  } = {}
+) => {
+  const positionsAudit = []
+
+  for (const id of ids) {
+    let end = Date.now()
+    let prevEnd = end
+    let serialRequestsCount = 0
+
+    while (true) {
+      const _res = await rService.getPositionsAudit(
+        null,
+        { auth, params: { id: [id], end, limit: 250 } }
+      )
+
+      const { res, nextPage } = (
+        Object.keys({ ..._res }).every(key => key !== 'nextPage')
+      )
+        ? { res: _res, nextPage: null }
+        : _res
+
+      prevEnd = end
+      end = nextPage
+
+      if (
+        Array.isArray(res) &&
+        res.length === 0 &&
+        nextPage &&
+        Number.isInteger(nextPage) &&
+        serialRequestsCount < 1
+      ) {
+        serialRequestsCount += 1
+
+        continue
+      }
+
+      serialRequestsCount = 0
+
+      if (
+        !Array.isArray(res) ||
+        res.length === 0
+      ) {
+        break
+      }
+
+      const resWithoutDuplicate = _filterDuplicate(positionsAudit, res)
+      positionsAudit.push(...resWithoutDuplicate)
+
+      if (
+        !Number.isInteger(nextPage) ||
+        (
+          resWithoutDuplicate.length === 0 &&
+          end === prevEnd
+        )
+      ) {
+        break
+      }
+    }
+  }
+
+  return positionsAudit
```

Comment:
And here just returning the last one would be fine, not need to return the complete array.
Just saw that then is filtered with _filterPositions, however this can be filtered here.
Also if position is closed there is no need to added to the audit ones.